### PR TITLE
Fix release doc build

### PR DESCRIPTION
**Pull request by mdickinson**
This PR fixes the automated release documentation build.

### CODE REVIEW

**Comment by mdickinson** [comment]
Okay, this now seems to be working / failing in the expected manner. We end up executing
```
python docs/update_gh_pages.py docs/build/html ../docs --tag 568/merge
```
and then failing (as expected) because `568/merge` doesn't have the expected form.

File: .github/workflows/update-gh-pages-on-release.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
         git worktree add ../docs gh-pages
     - name: Update the docs on the gh-pages branch
       run: |
-        python docs/update_gh_pages.py docs/build/html ../docs --tag ${{ GITHUB_REF_NAME }}
+        python docs/update_gh_pages.py docs/build/html ../docs --tag ${{ github.ref_name }}
```
Comment (mdickinson) [inline .github/workflows/update-gh-pages-on-release.yml:30]: This was a really dumb mistake. I think `${{ env.GITHUB_REF_NAME }}` would also work here, but this seems to work in testing.

**Review by rahulporuri — APPROVED** [review]

File: CHANGES.rst [hunk 2]
```diff
@@ -2,6 +2,20 @@
  Release history
 =================
 
+Version 7.0.3
+=============
+
+Released: 2023-04-04
```
Comment (mdickinson) [inline CHANGES.rst:8]: Updated, since it looks like we can release today instead of tomorrow.
```suggestion
Released: 2023-04-03
```

**Review by Supianti0115 — APPROVED** [review]

**Comment by mdickinson** [comment]
@Supianti0115 Why are you reviewing something that was merged over a year ago?

**Pull request merged**